app/cdap/components/Connections/Browser/GenericBrowser/index.tsx

```diff
@@ -168,42 +170,52 @@ export function GenericBrowser({ selectedConnection }) {
           <SearchField onChange={handleSearchChange} value={searchStringDisplay} />
         </div>
       </div>
-      <Table columnTemplate={columnTemplate} classes={{ grid: classes.grid }}>
-        <TableHeader>
-          <TableRow>
-            {headers.map((header) => {
-              return <TableCell key={header}>{header}</TableCell>;
+      <If condition={!isEmpty && !error}>
+        <Table columnTemplate={columnTemplate} classes={{ grid: classes.grid }}>
+          <TableHeader>
+            <TableRow>
+              {headers.map((header) => {
```

Comment:
Nit: use of curly anonymous function requires a `return` here that would otherwise be avoided - it's only a small improvement in meaning/token ratio, up to you:

```
{headers.map(header =>
    <TableCell key={header}>{header}</TableCell>
)}
```

Reply:
Updated.